Upgraded WPILib and Chopshoplib to 2021

andrewmartin5 opened the pull request:
upgraded versions of both wpilib and chopshoplib to their 2021 versions. Also updated imports and method names

andrewmartin5 committed:
Upgraded build and preferences to wpilib

updated build and preferences to wpilib 2021-opted in to simulating drivers station

andrewmartin5 committed:
fixed driver station simulation

added line to allow for driver station simulation

andrewmartin5 committed:
pushed to wrong branch so this is the upgrade to chopshoplib 2021.0.0

updated version name, as well as changing basic imports and command names

andrewmartin5 committed:
Fixed build

it builds now- that was an adventure

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -7,8 +7,8 @@
 
-import com.chopshop166.chopshoplib.DashboardUtils;
 import com.chopshop166.chopshoplib.RobotUtils;
+import com.chopshop166.chopshoplib.commands.CommandRobot;
```
Inherit from `CommandRobot` instead of `TimedRobot` on line 52, then on line 62 you can remove the `CommandRobot.` part of it

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -104,8 +105,6 @@ public void robotInit() {
 
         Shuffleboard.getTab("Shuffleboard").add("Autonomous", autoChooser);
 
-        DashboardUtils.logTelemetry();
```
This is also only fixed if you inherit from `CommandRobot` and on line 84 add `super.robotInit();`

msoucy commented on src/main/java/frc/robot/maps/FrancoisMap.java:
```diff
@@ -123,8 +123,8 @@ public PIDSparkMax shooterWheel() {
-            public SendableSpeedController spinner() {
-                SendableSpeedController motor = SendableSpeedController.wrap(controlPanel);
+            public SmartSpeedController spinner() {
+                SmartSpeedController motor = SmartSpeedController.wrap(controlPanel);
```
Line seems a bit redundant with the below line, just merge the two like what happens elsewhere

andrewmartin5 replied:
doesn't the control panel speed controller need to be defined up at the top of FrancoisMap so that we can use it the control panel gyro?

bot190 replied:
It is defined at the top of the robot map class. This is a wrapper class that we don't use for anything here and can be removed.

msoucy commented on src/main/java/frc/robot/subsystems/ControlPanel.java:
```diff
@@ -81,21 +81,21 @@ public ColorStates getTargetColor() {
-            default:
-                color = ColorStates.OTHER;
-                break;
+                case 'R':
```
Indentation changes?

andrewmartin5 replied:
pretty sure that was autoformat- should i leave it?

msoucy replied:
Meh fine

andrewmartin5 committed:
Some pull request fixes

robot.java now inherits commandrobot, robot properly initializes

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -82,6 +81,7 @@
     @Override
     public void robotInit() {
         Logger.configureLoggingAndConfig(this, false);
+        super.robotInit();
```
It should probably be the very first thing inside this function

msoucy commented:
Also make sure to check settings.gradle

msoucy commented:
[WPIlib 2021.2.1](https://github.com/wpilibsuite/allwpilib/releases/tag/v2021.2.1) is already released - please bump the version number in this PR so we don't need another one for it.

Also I recommend using the WPIlib "Manage Vendor Libraries > Update Offline" feature in VS Code, to make sure that all dependencies are up to date (CAN stuff in particular)

andrewmartin5 committed:
Updated to wpilib version 2021.2.1, as well as general cleanup

msoucy commented on src/main/java/frc/robot/maps/FrancoisMap.java:
```diff
@@ -122,11 +122,6 @@ public PIDSparkMax shooterWheel() {
-            @Override
-            public SmartSpeedController spinner() {
-                SmartSpeedController motor = SmartSpeedController.wrap(controlPanel);
-                return motor;
```
This should not have been removed - just combined into one line

andrewmartin5 committed:
I think i fixed it?

re-added some lines that were removed last commit and combined a couple of them into one line
